packages/firestore/android/src/reactnative/java/io/invertase/firebase/firestore/ReactNativeFirebaseFirestoreSerialize.java

```diff
@@ -404,6 +405,36 @@ private static WritableArray buildTypeMap(Object value) {
       return typeArray;
     }
 
+    // VectorValue – detect via reflection to avoid compile-time dependency on newer SDKs
```

Comment:
Notes:
- firebase-android-sdk released Vector value support in BoM 33.2.0 on August 22 2024: https://firebase.google.com/support/release-notes/android#firestore_v25-1-0
- firebase-ios-sdk released Vector value support in v11.1.0 on August 19 2024: https://firebase.google.com/support/release-notes/ios#cloud-firestore_11

Our most recent breaking change release here was v33.0.0 which included firebase-android-sdk 34.0.0 and firebase-ios-sdk 12.0.0

So: it is a safe and non-breaking change to immediately begin relying on these concrete types here with no need for reflection, and the code should be changed to that simpler style